### PR TITLE
Use squiggly heredocs instead of strip_heredoc.

### PR DESCRIPTION
**Pull request by troessner**
I'd also like to suggest to rename *all* heredoc delimiters we have to HERE so it's consistent everywhere (in a separate PR though). Right now it's EOF, EOS, SRC, FOO and god knows what else ;)

### CODE REVIEW

**Comment by mvz** [comment]
* Yes to squiggly heredocs if supported by all supported Rubies
* No to having all delimiters be `HERE`. Let's use a format-descriptive delimiter instead as suggested by the `Naming/HeredocDelimiterNaming` cop (currently disabled by our config), and the discussion in https://github.com/troessner/reek/pull/1259#discussion_r139307715.

File: spec/reek/report/code_climate/code_climate_report_spec.rb [hunk 1]
```diff
@@ -26,7 +26,7 @@
     let(:source) { 'def simple(a) a[3] end' }
 
     it 'prints smells as json' do
-      expected = <<-EOS.strip_heredoc.delete("\n")
+      expected = <<~HERE.delete("\n")
```
Comment (mvz) [inline spec/reek/report/code_climate/code_climate_report_spec.rb]: I really wonder why we escape all the double quotes below here?

**Comment by mvz** [comment]
RuboCop may be able to autocorrect all the strip_heredocs if you change the config?

**Comment by mvz** [comment]
I meant what @Drenmi said :slightly_smiling_face: .

**Comment by troessner** [comment]
Makes sense, updated ;=

**Pull request merged**